src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -510,7 +510,19 @@ void ScenesServer::GroupWillBeRemoved(FabricIndex aFabricIx, EndpointId aEndpoin
     SceneTable * sceneTable = scenes::GetSceneTableImpl(aEndpointId);
     VerifyOrReturn(nullptr != sceneTable);
 
-    MakeSceneInvalid(aEndpointId);
+    chip::GroupId * currentGroup;
+    Attributes::CurrentGroup::Get(aEndpointId, currentGroup);
+
+    chip::SceneId * currentScene;
+    Attributes::CurrentScene::Get(aEndpointId,currentScene);
+
+    //Validate if the Current Group = RemovedGroup or check if CurrentScene is in RemovedGroup before invalidating the scene.
+    SceneTableEntry entry;
+    if(aGroupId == *currentGroup || sceneTable->GetSceneTableEntry(aFabricIx,SceneStorageId(*currentScene,*currentGroup),entry) == CHIP_NO_ERROR)
```

Comment:
Why does it matter whether `aGroupId == currentGroup`?

Why does it matter whether there is a scene entry for the _current_ bits?

Reply:
My understanding is that this impacts the SceneValid attribute of the scene server.

1.4.8.4. SceneValid Attribute

> The SceneValid attribute indicates whether the state of the server corresponds to that associated
> with the CurrentScene and CurrentGroup attributes. TRUE indicates that these attributes are valid,
> FALSE indicates that they are not valid.
> Before a scene has been stored or recalled, this attribute is set to FALSE. After a successful
> StoreScene or RecallScene command it is set to TRUE. If, after a scene is stored or recalled, the state
> of the server is modified, this attribute is set to FALSE.

Thus, if the current group is removed, SceneValid must be set to false.

Reply:
I understand what the right behavior is.  Can you please explain to me why this code is implementing the right behavior?

Reply:
As it stands, the yaml scene tests are currently disabled on the CI as the feature is still provisionary. We can only test it locally but nothing will be implemented by default until many of the trailing issues are solved. There are still open issues about completing the yaml tests, implementing missing cluster callbacks, updating scenes cluster to respect latest spec changes to be made before we can have scene tests in the CI and expect it to pass.